app/src/androidTest/java/org/mozilla/focus/activity/robots/SettingsGeneralMenuRobot.kt

```diff
@@ -5,28 +5,81 @@
 package org.mozilla.focus.activity.robots
 
 import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
 import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.espresso.matcher.ViewMatchers.isChecked
 import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isNotChecked
+import androidx.test.espresso.matcher.ViewMatchers.withId
 import androidx.test.espresso.matcher.ViewMatchers.withText
-import org.mozilla.focus.helpers.TestHelper.pressBackKey
+import androidx.test.uiautomator.UiSelector
+import junit.framework.TestCase.assertTrue
+import org.mozilla.focus.R
+import org.mozilla.focus.helpers.TestHelper.appName
+import org.mozilla.focus.helpers.TestHelper.mDevice
+import org.mozilla.focus.helpers.TestHelper.waitingTime
 
 class SettingsGeneralMenuRobot {
+    companion object {
+        const val DEFAULT_APPS_SETTINGS_ACTION = "android.settings.MANAGE_DEFAULT_APPS_SETTINGS"
+    }
 
     fun verifyGeneralSettingsItems() {
         defaultBrowserSwitch.check(matches(isDisplayed()))
         switchToLinkToggleButton.check(matches(isDisplayed()))
     }
 
-    class Transition {
-        fun goBack(interact: SettingsRobot.() -> Unit): SettingsRobot.Transition {
-            pressBackKey()
+    fun clickSetDefaultBrowser() {
+        defaultBrowserSwitch
+                .check(matches(isDisplayed()))
+                .perform(click())
+    }
+
+    fun verifyAndroidDefaultAppsMenuAppears() {
+        intended(IntentMatchers.hasAction(DEFAULT_APPS_SETTINGS_ACTION))
+    }
+
+    fun verifyOpenWithDialog() {
+        openWithDialogTitle.waitForExists(waitingTime)
+        assertTrue(openWithDialogTitle.exists())
+        assertTrue(openWithList.exists())
+    }
 
-            SettingsRobot().interact()
-            return SettingsRobot.Transition()
-        }
+    fun selectAlwaysOpenWithFocus() {
+        mDevice.findObject(UiSelector().text(appName)).click()
+        mDevice.findObject(
+                UiSelector()
+                        .resourceId("android:id/button_always")
+                        .enabled(true)
+        ).click()
+    }
+
+    fun verifySwitchIsOn(checked: Boolean) {
```

Comment:
I would maybe rename to verifySwitchIsToggled

Reply:
done